### PR TITLE
Add CARLA model

### PR DESCRIPTION
**Pull request by mockoocy**
Includes a 1D ResNet implementation to serve as the backbone for CARLA.  Created `FeatureExtractor` protocol for that case, as possibly these could be reused for some other approaches.
Implementation is mostly faithful to the original CARLA repo: https://github.com/zamanzadeh/CARLA

### CODE REVIEW

**Comment by mockoocy** [comment]
I've pushed a change in the validation metric. There's an early stop in place w.r.t. validation loss in the classification stage of the training. This is to replace choosing model with the best F1 value across all epochs from the reference implementation, as we may not necessarily have access to the anomaly labels in the validation set; as is the case with the energy dataset.

File: src/pyclad/models/contrastive/carla.py [hunk 1]
```diff
@@ -0,0 +1,588 @@
...
+                optimizer.step()
+                total_loss += loss.item() * batch_size
+                n_samples += batch_size
+            train_mean_loss = total_loss / max(1, n_samples)
+
+            module.eval()
+            val_total = 0.0
+            val_samples = 0
+            with torch.no_grad():
+                for start in range(0, n_val, self._batch_size):
```
Comment (Nyderx) [inline src/pyclad/models/contrastive/carla.py:511]: training inner loop (lines ~469–493) and validation inner loop (~500–522) are near-identical: same reshape math, same _classification_loss call, same loss accumulation. The only differences are the index source (perm[start:end] vs torch.arange(start, end)) and the no_grad/optimizer wrapper. Any fix to
  the loss computation or reshape must be applied twice. Extract a shared _eval_batch(module, windows, nn_idx, fn_idx, batch_idx) helper.

Comment (mockoocy) [reply src/pyclad/models/contrastive/carla.py:511]: Done. I've also rebased this PR's branch.

**Review by Nyderx — APPROVED** [review]
